app/src/main/java/org/pesmypetcare/mypetcare/services/VetVisitsAdapter.java

```diff
@@ -0,0 +1,88 @@
+import java.util.concurrent.ExecutionException;
+
+/**
+ * @author Xavier Campos
+ */
+public class VetVisitsAdapter implements VetVisitsManagerService {
+    @Override
+    public List<VetVisit> findVetVisitsByPet(User user, Pet pet) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        ArrayList<VetVisit> appVisits = new ArrayList<>();
+        List<org.pesmypetcare.usermanagerlib.datacontainers.VetVisit> libraryVisit =
+            ServiceLocator.getInstance().getPetCollectionsManagerClient().getAllVetVisits(accessToken, owner, petName);
+        for (org.pesmypetcare.usermanagerlib.datacontainers.VetVisit visit : libraryVisit) {
+            appVisits.add(new VetVisit(DateTime.Builder.buildFullString(visit.getKey()), visit.getBody().getAddress(),
+                visit.getBody().getReason()));
+        }
+        return appVisits;
+    }
+
+    @Override
+    public void createVetVisit(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+
+        VetVisitData visitData = new VetVisitData(vetVisit.getReason(), vetVisit.getAddress());
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit libraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(vetVisit.getVisitDate().toString(), visitData);
+
+        ServiceLocator.getInstance().getPetManagerClient().addFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, libraryVisit.getKey(), libraryVisit.getBodyAsMap());
+    }
+
+    @Override
+    public void deleteVetVisit(User user, Pet pet, VetVisit vetVisit) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        VetVisitData visitData = new VetVisitData(vetVisit.getReason(), vetVisit.getAddress());
+        org.pesmypetcare.usermanagerlib.datacontainers.VetVisit libraryVisit =
+            new org.pesmypetcare.usermanagerlib.datacontainers.VetVisit(vetVisit.getVisitDate().toString(), visitData);
+        ServiceLocator.getInstance().getPetManagerClient().deleteFieldCollectionElement(accessToken, owner, petName,
+            PetData.VET_VISITS, libraryVisit.getKey());
+    }
+
+    @Override
+    public void updateVetVisitKey(User user, Pet pet, String newDate, DateTime visitDate) throws ExecutionException, InterruptedException {
+        String accessToken = user.getToken();
+        String owner = user.getUsername();
+        String petName = pet.getName();
+        VetVisitData vetVisitData = ServiceLocator.getInstance().getPetCollectionsManagerClient().getVetVisit
+            (accessToken, owner, petName, visitDate.toString());
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['(' should be on the previous line.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care/pullRequest?prid=5493224)